srml/system/src/lib.rs

```diff
@@ -14,8 +14,74 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! System manager: Handles lowest level stuff like depositing logs, basic set up and take down of
-//! temporary storage entries, access to old block hashes.
+//! # System module
+//! 
+//! The system module provides low-level access to core types and cross-cutting utilities.
+//! It acts as the base layer, for other SRML modules, to interact with the Substrate framework components.
+//! To use it in your module, you need to implement the system [`Trait`].
+//! 
+//! ## Overview
+//! 
+//! The system module defines the core data types used in Substrate runtime. 
+//! It also provides several utility functions ([`Module`]) for other modules.
+//! 
+//! In addition, it manages the storage items for extrinsics data, indexes, event record and digest items, 
+//! among other things that support the execution of the current block.
+//! 
+//! It also handles low level tasks like depositing logs, basic set up and take down of
+//! temporary storage entries and access to previous block hashes.
+//! 
+//! ## Interface
+//! 
+//! ### Dispatchable functions
+//! 
+//! The system module does not implement any dispatchable functions.
+//! 
+//! ### Public functions - [`Module`]
+//! 
+//! ## Usage
+//! 
+//! ### Prerequisites
+//! 
+//! Import the system module and derive your module configuration trait from the `system` trait.
+//! 
+//! ### Example - get random seed for the current block
+//! 
+//! ```ignore
+//! use support::{decl_module, dispatch::Result};
+//! use system::ensure_signed;
+//! 
+//! pub trait Trait: system::Trait {}
+//! 
+//! decl_module! {
+//! 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+//! 		pub fn get_time(origin) -> Result {
```

Comment:
Why is this called `get_time`? Maybe something different `my_cool_function` or so, because you don't return any time.